### ISSUE TITLE
add some sort of locking support to EntityManagerImpl

### ISSUE THREAD

**Issue by mhgrove** [mhgrove/Empire#9]
right now we don't do anything other than validate the the EntityManager is not closed and that you're not locking on something that is not a managable entity.

in the short term we might want to throw a persistence exception so users know that the lock was not obtained rather silently ignoring the request.

**Comment by tr-jsantoru** [comment]
@mhgrove this is an issue from 2010, but was wondering if there's currently any support in Empire (or any plans) for any kind of locking, specifically optimistic locking?

**Comment by mhgrove** [comment]
No, the support has not been added. I still plan on doing it, but I don't have any timeline I could give you for that.

**Comment by tr-jsantoru** [comment]
No problem. Thanks for the prompt response, much appreciated.
